param_passing/proposal.md

```diff
@@ -0,0 +1,388 @@
+    <include>
+      <uri>model://base_robot</uri>
+
+      <experimental:params>
+        <visual element_id="chassis::lidar_visual" action="remove"/>
+        <sensor element_id="chassis::lidar" action="remove"/>
+
+        <sensor element_id="chassis::camera" action="add">
+          <plugin name="camera_depth_sensor" filename="libcamera_depth_sensor.so"/>
+        </sensor>
+
+        <visual element_id="chassis::camera_visual" action="add">
+          <pose>0.5 0.02 0 0 0</pose>
+          <geometry>
+              <box>
+                  <size>0.02 0.02 0.02</size>
+              </box>
+          </geometry>
+        </visual>
+
+        <sensor element_id="top::camera" action="modify">
+          <pose>0 1 0 0 0 0</pose>
+          <update_rate>60</update_rate>
+        </sensor>
+
+        <visual element_id="top::camera_visual">
+          <geometry action="replace">
+            <sphere>
+              <radius>0.05</radius>
+            </sphere>
+          </geometry>
+          <material action="modify">
+            <ambient>0 1 0 1</ambient>
+          </material>
+        </visual>
+
+      </experimental:params>
+    </include>
+  </model>
+</sdf>
+```
+
+In `//include/experimental:params`, the action can be either listed in the
+element identifier or listed individually in the children element(s).
+For example,
+
+```xml
+<sensor element_id="top::camera" action="modify">
+  <pose>0 1 0 0 0 0</pose>
+  <update_rate>60</update_rate>
+</sensor>
+```
+
+The action is defined at the top level where the `element_id` is, which means
+that all children listed in this element will be modified. If the user wants to
+do several different actions to the children elements then the user can specify
+the action for each child element individually. For instance,
+
+```xml
+<visual element_id="top::camera_visual">
+  <geometry action="replace">
+    <sphere>
+      <radius>0.05</radius>
+    </sphere>
+  </geometry>
+  <material action="modify">
+    <ambient>0 1 0 1</ambient>
+  </material>
+</visual>
+```
+
+If the original model has several corresponding elements, it will be up to the
+user to update all appropriately. For example, if a lidar has a `//visual` and
+`//sensor` element and the user would like to remove the lidar information, then
+both will need to be listed for removal:
+
+```xml
+<visual element_id="chassis::lidar_visual" action="remove"/>
+<sensor element_id="chassis::lidar" action="remove"/>
+```
+
+When a user wants to `add` an element that does not exist in the original model,
+then the `add` action can be used similarly as follows:
+
+```xml
+<visual element_id="chassis::camera_visual" action="add">
+  <pose>0.5 0.02 0 0 0</pose>
+  <geometry>
+      <box>
+          <size>0.02 0.02 0.02</size>
+      </box>
+  </geometry>
+</visual>
+```
+
+In this example, the `//visual` element named "camera_visual" will be added as a
+child of `//link[@element_id=’chassis’]`. A check will be performed to ensure
```

Comment:
For whatever reason, `sdf_web_app` is struggling to render this part; perhaps due to the odd quotes?
http://sdformat.org/tutorials?tut=param_passing_proposal&branch=param_passing#example-1
![image](https://user-images.githubusercontent.com/26719449/106182072-5bd54d80-616c-11eb-8a20-36564684d38c.png)

Reply:
Fixed 7c8d6438fab8772a89119f927cdc145cf2948772